buildSrc/plugins/src/main/kotlin/org/jetbrains/kotlin/KonanTestSuite.kt

```diff
@@ -0,0 +1,139 @@
+package org.jetbrains.kotlin
+
+import java.io.PrintWriter
+import java.io.StringWriter
+
+enum class TestStatus {
+    PASSED,
+    FAILED,
+    ERROR,
+    SKIPPED
+}
+
+data class TestResult(val status: TestStatus, val comment: String = "")
+
+data class Statistics(
+        var passed: Int = 0,
+        var failed: Int = 0,
+        var error: Int = 0,
+        var skipped: Int = 0) {
+
+    val total: Int
+        get() = passed + failed + error + skipped
+
+    fun pass(count: Int = 1) { passed += count }
+
+    fun skip(count: Int = 1) { skipped += count }
+
+    fun fail(count: Int = 1) { failed += count }
+
+    fun error(count: Int = 1) { error += count }
+
+    fun add(other: Statistics) {
+        passed += other.passed
+        failed += other.failed
+        error += other.error
+        skipped += other.skipped
+    }
+}
+
+open class KonanTestSuite(val name: String, val statistics: Statistics) {
+
+    open inner class KonanTestCase(val name: String) {
+        open fun start() {}
+
+        open fun pass(): TestResult {
+            statistics.pass()
+            return TestResult(TestStatus.PASSED)
+        }
+
+        open fun fail(e: RuntimeException): TestResult {
+            statistics.fail()
+            println(e.message)
+            return TestResult(TestStatus.FAILED, "Exception: ${e.message}. Cause: ${e.cause?.message}")
+        }
+
+        open fun error(e: Exception): TestResult {
+            statistics.error()
+            return TestResult(TestStatus.ERROR, "Exception: ${e.message}. Cause: ${e.cause?.message}")
+        }
+
+        open fun skip(): TestResult {
+            statistics.skip()
+            return TestResult(TestStatus.SKIPPED)
+        }
+    }
+
+    open fun createTestCase(name: String) = KonanTestCase(name)
+
+    open fun start() {}
```

Comment:
Maybe skip body and require subclass to implement?

Reply:
By default KonanTestSuite is used by the standalone cli-exectution, while subclasses are used for the TeamCity

Reply:
I'd suggest to stay at this phase (naive and working groovy to kotlin conversion) to let me continue with refactoring and modification this part of code.